src/CompileTime.hx

```diff
@@ -64,6 +70,17 @@ class CompileTime
         return toExpr(obj);
     }
 
+    #if yaml
+    macro public static function parseYamlFile(path:String) {
+      var content = loadFileAsString(path);
+      var data = Yaml.parse(content, Parser.options().useObjects());
+      var s = haxe.Json.stringify(data);
+      var json = haxe.Json.parse(s);
+      return toExpr(json);
```

Comment:
Because Json parse do a better job for handling date object format.
Without json parsing i get this error for date:
/usr/share/haxe/std/haxe/macro/Context.hx:295: characters 9-37 : Unsupported value 2001-01-23 00:00:00